src/browser/engine/protocol.ts

```diff
@@ -30,52 +83,65 @@ const protocolHandler = async (
   if (!extension) return callback();
 
   const { src, backgroundPage: { name, html } } = extension;
-  const headers = {};
 
-  // Todo : Hack to get extension ID,
-  // revert to hostname when this (https://github.com/getstation/electron-chrome-extension/commit/8f8d37e13c47611ce9c8b184775a68fa52fc883d) is reverted too
-  const splitted = src.split('/');
-  const ecxId = splitted[splitted.length - 2];
+  const manifestPath = join(src, 'manifest.json');
+  const manifestFile = await readFileSync(manifestPath, 'utf-8');
+  const manifest = JSON.parse(manifestFile);
 
-  // Set Content Security Policy for Chrome Extensions
-  if (ECx.isLoaded(ecxId)) {
-    const manifestPath = join(src, 'manifest.json');
-    const manifest = await readFileSync(manifestPath, 'utf-8');
+  const headers = new Map();
 
-    const manifestContentSecurityPolicy = JSON.parse(manifest).content_security_policy;
-    const contentSecurityPolicy = manifestContentSecurityPolicy ? manifestContentSecurityPolicy : defaultContentSecurityPolicy;
+  // Always send CORS
+  // refs:
+  // https://cs.chromium.org/chromium/src/extensions/browser/extension_protocols.cc?l=524
+  // https://cs.chromium.org/chromium/src/extensions/browser/extension_protocols.cc?l=330
+  // https://cs.chromium.org/chromium/src/extensions/browser/extension_protocols.cc?l=1017
+  headers.set('access-control-allow-origin', '*');
 
-    headers['content-security-policy'] = contentSecurityPolicy;
-  }
+  // Set Content Security Policy for Chrome Extensions
```

Comment:
You have a reference to Chromium or extension documentation for this?

Reply:
added